### PR TITLE
Add possibility to pass secrets via files

### LINKED ISSUE #30: Feature request: Read secrets from files
I'm deploying this service on [NixOS](https://nixos.org) as a systemd unit and I manage secrets using [sops](https://github.com/getsops/sops). A common practice in this space is for applications to read the location of a file containing a secret from the system environment, instead of reading the secret itself from the environment.

For this reason I'd like to be able to configure `DYNDNS_SERVER_PASSWORD_FILE` environment variable that contains the path to a file with the password. The application should inspect this variable in case `DYNDNS_SERVER_PASSWORD` is not set and try to read the password from the file specified that way. Same for `CLOUDFLARE_API_TOKEN`: The application should try to read the token from a file specified via `CLOUDFLARE_API_TOKEN_FILE` in case `CLOUDFLARE_API_TOKEN` is not set.

### PR DESCRIPTION
**Pull request by britter**
Resolves #30

### CODE REVIEW

**Comment by britter** [comment]
Tested this in my homelab and it looks like it's working.

File: main.go [hunk 1]
```diff
@@ -272,3 +272,22 @@ func startPollServer(out chan<- *net.IP, localIp *net.IP) {
...
+	passwordFilePath := os.Getenv(envName + "_FILE")
+	if passwordFilePath != "" {
+		content, err := os.ReadFile(passwordFilePath)
+		if err != nil {
+			slog.Error("Failed to read DynDns server password from file", logging.ErrorAttr(err))
+		} else {
+			password = string(content)
+		}
+	}
+	return password
```
Comment (cromefire) [inline main.go]: It probably makes sense to stick with the "secret" terminology.

File: main.go [hunk 2]
```diff
@@ -272,3 +272,22 @@ func startPollServer(out chan<- *net.IP, localIp *net.IP) {
...
+
+	if password != "" {
+		return password
+	}
+
+	passwordFilePath := os.Getenv(envName + "_FILE")
+	if passwordFilePath != "" {
+		content, err := os.ReadFile(passwordFilePath)
+		if err != nil {
+			slog.Error("Failed to read DynDns server password from file", logging.ErrorAttr(err))
```
Comment (cromefire) [inline main.go]: And this message should be more generic as well logging that it's unable to read the file specified in $envName.

**Review by cromefire — COMMENTED** [review]
Also the docker samples in the readme should be updated accordingly to make use of secrets.

**Comment by britter** [comment]
@cromefire I addressed the two code comments.

> Also the docker samples in the readme should be updated accordingly to make use of secrets.

I don't use docker that much so I don't exactly know how to pass secrets via files. Would you mind adding this yourself on top of my changes?

**Comment by cromefire** [comment]
Yeah was planning to, but I probably won't get to it today, I'll try and add it in the coming days, when I test it locally anyways.

**Comment by britter** [comment]
Hey @cromefire did you have a chance to look into this yet? Would love to have a release with this included.

**Comment by cromefire** [comment]
No I didn't have any time to look at it or test it yet. Time's currently very scarce for me, I hope I do get to it in the next few days, but thanks for the reminder.

**Comment by britter** [comment]
Hey @cromefire I've given it a shot and added an example for how to use the new feature with docker compose secrets. Let me know what you think!

**Review by cromefire — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -153,6 +153,37 @@ Now we could configure the FRITZ!Box
...
+      - DYNDNS_SERVER_BIND=:8080
+      - CLOUDFLARE_API_KEY_FILE=/run/secrets/cloudflare_api_key
+      - CLOUDFLARE_ZONES_IPV4=test.example.com
+      - CLOUDFLARE_ZONES_IPV6=test.example.com
+    secrets:
+      - cloudflare_api_key
+
+secrets:
+  cloudflare_api_key:
+    file: ./cloudflare_api_key.txt
```
Comment (cromefire) [inline README.md]:
```suggestion
      - CLOUDFLARE_API_TOKEN_FILE=/run/secrets/cloudflare_api_token
      - DYNDNS_SERVER_PASSWORD_FILE=/run/secrets/fb_server_password
      - CLOUDFLARE_ZONES_IPV4=test.example.com
      - CLOUDFLARE_ZONES_IPV6=test.example.com
    secrets:
      - cloudflare_api_token
      - fb_server_password

secrets:
  cloudflare_api_token:
    file: ./cloudflare_api_token.txt
  fb_server_password:
    file: ./fb_server_password.txt
```

Also add FB password to the example and use the CF Token, not the deprecated key.

File: main.go [hunk 4]
```diff
@@ -272,3 +272,22 @@ func startPollServer(out chan<- *net.IP, localIp *net.IP) {
...
+		return secret
+	}
+
+	passwordFilePath := os.Getenv(envName + "_FILE")
+	if passwordFilePath != "" {
+		content, err := os.ReadFile(passwordFilePath)
+		if err != nil {
+			slog.Error("Failed to read secret from file "+passwordFilePath, logging.ErrorAttr(err))
+		} else {
+			secret = string(content)
```
Comment (cromefire) [inline main.go]:
```suggestion
			secret = strings.TrimSuffix(strings.TrimSuffix(string(content), "\r\n"), "\n")
```
Some ways to put the secret into the file (echo without `-n`, nano without `-L`, probably more), leave a newline at the end, this should strip it to not cause any unexplained errors.

**Review by cromefire — COMMENTED** [review]
Works in general, but I'd have 2 smaller requests in addition to the 2 comments (but those I'll simply review and don't need to retest, so I promise it'll be merged soon after and I'll trigger a release then!):
- Please make a note on the 2 secret variables (in the variable list in the readme) that it's recommended to use files/secrets if possible
- Please remove the secret variables from the docker files (or add the `_FILE` suffix)

Also if you fancy (I leave that up to you) maybe an info level log in the `readSecret` function in the case that the env is set, recommending to use files instead (maybe with a link to the readme secrets section: `https://github.com/cromefire/fritzbox-cloudflare-dyndns?tab=readme-ov-file#passing-secrets`)?

You can also put yourself into the `CONTRIBUTORS.md` if you want.

**Comment by cromefire** [comment]
`v1.3.0` has been created, binaries are attached and containers are built

**Pull request merged**